sceptre/cli/dump.py

```diff
@@ -53,3 +55,55 @@ def dump_config(ctx, path):
     else:
         for config in output:
             write(config, output_format)
+
+
+@dump_group.command(name="template")
+@click.argument("path")
+@click.option(
+    "-n",
+    "--no-placeholders",
+    is_flag=True,
+    help="If True, no placeholder values will be supplied for resolvers that cannot be resolved.",
+)
+@click.pass_context
+@catch_exceptions
+def dump_template(ctx, no_placeholders, path):
+    """
+    Prints the template used for stack in PATH.
+    \f
+
+    :param path: Path to execute the command on.
+    :type path: str
+    """
+    context = SceptreContext(
+        command_path=path,
+        command_params=ctx.params,
+        project_path=ctx.obj.get("project_path"),
+        user_variables=ctx.obj.get("user_variables"),
+        options=ctx.obj.get("options"),
+        output_format=ctx.obj.get("output_format"),
+        ignore_dependencies=ctx.obj.get("ignore_dependencies"),
+    )
+
+    plan = SceptrePlan(context)
+
+    execution_context = (
+        null_context() if no_placeholders else use_resolver_placeholders_on_error()
+    )
+    with execution_context:
+        responses = plan.dump_template()
+
+    output = []
+    for stack, template in responses.items():
+        if template is None:
+            logger.warning(f"{stack.external_name} does not exist")
+        else:
+            output.append({stack.external_name: template})
+
+    output_format = "json" if context.output_format == "json" else "yaml"
+
+    if len(output) == 1:
+        write(output[0][stack.external_name], output_format)
+    else:
+        for template in output:
+            write(template, output_format)
```

Comment:
It strikes me as odd that the output format is fundamentally different if it's just one stack vs. if it's numerous stacks.

Reply:
It makes sense if you use it, and, again, is what you and I agreed on already after lengthy discussions for `dump config`. In the case of a single stack—most of the time—you don't want to see the stack names in this output; you just want the yaml output. In the case of a stack group however, you need to know which stack goes with which doc.

HOWEVER, please note that in the implementation of Khai's feature, I have changed this back to multi-doc yaml output without the stack names. That's because the .dump structure that is created allows us to know which stack goes with which doc.

Reply:
Closing this because it's not relevant after I merged Part 2 into here.